### DISCUSSION TITLE
Classifier free guidance paper and code, performance bump

### DISCUSSION THREAD

**Discussion by jjohare** [oobabooga/text-generation-webui#2989] [Ideas]
As per the feature request to Llama.cpp

https://github.com/ggerganov/llama.cpp/issues/2083

and this tweet and paper, ["Stay on topic with Classifier-Free Guidance"](https://twitter.com/Vermeille_/status/1675664118500454400)

 ... "brings improvements equivalent to a model with twice the parameter-count" (with no retraining needed). 
 
 Possible moving [towards the transformers library](https://github.com/huggingface/transformers/issues/24536). Opening a discussion here so it doesn't get missed.

**Comment by jjohare** [comment]
In the wild implementation for GGML/MTP-7B
https://twitter.com/apage43/status/1676416243652505601
https://github.com/apage43/mptgen

**Reply by tmm1** [reply]
https://github.com/apage43/mptgen/commit/e9cedaa4d05377190bc51c305df2ef982101a0de

**Comment by FartyPants** [comment]
It's already in tranformers PR. The correct action is "wait"

Also the "performance bump" is kind of deceiving. Focus to tasks goes up (especially for small models), speed of generating goes down.
